test/goaway_test.go

```diff
@@ -816,8 +800,40 @@ func (s) TestClientSendsAGoAway(t *testing.T) {
 			}
 		}
 	}()
+	defer lis.Close()
+
+	cc, err := grpc.Dial(lis.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("error dialing: %v", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	//go func() {
+	//	for {
+	//		f, err := ct.fr.ReadFrame()
+	//		if err != nil {
+	//			errCh <- fmt.Errorf("error reading frame: %v", err)
+	//			return
+	//		}
+	//		switch fr := f.(type) {
+	//		case *http2.GoAwayFrame:
+	//			fr = f.(*http2.GoAwayFrame)
+	//			if fr.ErrCode == http2.ErrCodeNo {
+	//				t.Logf("GoAway received from client")
+	//				close(goAwayReceived)
+	//				return
+	//			}
+	//		default:
+	//			t.Errorf("server tester received unexpected frame type %T", f)
+	//			errCh <- fmt.Errorf("server tester received unexpected frame type %T", f)
+	//			close(errCh)
+	//		}
+	//	}
+	//}()
+	cc.WaitForStateChange(ctx, connectivity.Connecting)
```

Comment:
Should be `Ready` now.

Reply:
@dfawley `WaitForStateChange` waits until the connectivity.State of ClientConn changes from sourceState or ctx expires.  So aren't we already waiting for conn to be READY here as it would wait until state changes from Connecting?

Reply:
You are correct.  Instead you should probably be using `AwaitState(Ready)`:

https://github.com/grpc/grpc-go/blob/master/internal/testutils/state.go#L58

Reply:
Updated.